app/controllers/concerns/verify_profile_concern.rb

```diff
@@ -0,0 +1,32 @@
+module VerifyProfileConcern
+  private
+
+  def profile_or_verify_profile_path
+    public_send "#{profile_or_verify_profile_route}_path"
+  end
+
+  def profile_or_verify_profile_url
+    public_send "#{profile_or_verify_profile_route}_url"
+  end
+
+  def profile_or_verify_profile_route
+    return 'profile' if idv_context? || profile_context?
+    return 'profile' unless current_user.decorate.pending_profile_requires_verification?
+    verify_profile_route
+  end
+
+  def verify_profile_route
+    decorated_user = current_user.decorate
+    if decorated_user.needs_profile_phone_verification?
+      flash[:notice] = t('profile.index.verification.instructions')
+      return 'verify_profile_phone'
+    end
+    return 'verify_profile' if decorated_user.needs_profile_usps_verification?
+    'profile'
```

Comment:
I don't think an explicit exception is necessary. The final `else` will be `nil` if we remove this last line. If somehow this `nil` is reached, an error will be thrown automatically. As long as we have robust tests, it should protect us from someone breaking this logic.